### PR TITLE
Fixes from master to angular

### PR DESCRIPTION
**Pull request by alla-blinkova**

### CODE REVIEW

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UISelectAssert.java [hunk 1]
```diff
@@ -15,6 +15,7 @@
 import static com.epam.jdi.light.asserts.core.SoftAssert.*;
 import static com.epam.jdi.light.settings.JDISettings.*;
 import static com.epam.jdi.tools.EnumUtils.*;
+import static com.epam.jdi.tools.LinqUtils.*;
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UISelectAssert.java:18]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.tools.LinqUtils'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 23
- Added 12
           

Complexity increasing per file
==============================
- jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/BaseTable.java  13
- jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565&bid=18266460)

File: jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/asserts/TextAreaAssert.java [hunk 2]
```diff
@@ -7,7 +7,10 @@
 import org.hamcrest.Matcher;
 import org.hamcrest.Matchers;
 
+import java.util.List;
+
 import static com.epam.jdi.light.asserts.core.SoftAssert.*;
+import static org.hamcrest.Matchers.*;
```
Comment (codacy-review) [inline jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/asserts/TextAreaAssert.java:13]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java [hunk 3]
```diff
@@ -4,6 +4,8 @@
 import org.aspectj.lang.annotation.*;
 
 import static com.epam.jdi.light.actions.ActionHelper.*;
+import static com.epam.jdi.light.common.Exceptions.*;
+import static com.epam.jdi.light.settings.WebSettings.*;
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java:8]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.settings.WebSettings'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package com.epam.jdi.light.asserts.generic;
+
+import com.epam.jdi.light.common.JDIAction;
+import com.epam.jdi.light.elements.complex.IMultiSelector;
+import org.hamcrest.Matcher;
+
+import java.util.List;
+
+import static com.epam.jdi.light.asserts.core.SoftAssert.*;
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java:9]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.asserts.core.SoftAssert'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import static org.hamcrest.Matchers.*;
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java:10]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/TableSteps.java [hunk 6]
```diff
@@ -13,9 +13,10 @@
 import static com.epam.jdi.light.elements.complex.table.TableMatcher.hasValue;
 import static com.epam.jdi.light.elements.complex.table.TableMatcher.*;
 import static com.epam.jdi.light.elements.init.entities.collection.EntitiesCollection.*;
+import static com.epam.jdi.tools.LinqUtils.*;
+import static com.epam.jdi.tools.PrintUtils.*;
```
Comment (codacy-review) [inline jdi-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/TableSteps.java:17]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.tools.PrintUtils'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java [hunk 7]
```diff
@@ -4,6 +4,8 @@
 import org.aspectj.lang.annotation.*;
 
 import static com.epam.jdi.light.actions.ActionHelper.*;
+import static com.epam.jdi.light.common.Exceptions.*;
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java:7]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.common.Exceptions'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-bdd-tests/src/test/java/cucmberTests/stepdefs/UserStepdefs.java [hunk 8]
```diff
@@ -16,27 +13,10 @@
 import static io.github.com.entities.Users.*;
 import static io.github.com.pages.Header.*;
 import static java.util.Arrays.*;
-import static org.testng.Assert.*;
+import static org.junit.Assert.*;
```
Comment (codacy-review) [inline jdi-bdd-tests/src/test/java/cucmberTests/stepdefs/UserStepdefs.java:16]: Codacy Issue found: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-bdd/src/main/java/com/epam/jdi/bdd/JDISteps.java [hunk 9]
```diff
@@ -14,8 +14,8 @@
 import static com.epam.jdi.light.elements.init.UIFactory.*;
 import static com.epam.jdi.light.elements.init.entities.collection.EntitiesCollection.*;
 import static com.epam.jdi.light.settings.WebSettings.*;
+import static org.hamcrest.MatcherAssert.*;
```
Comment (codacy-review) [inline jdi-bdd/src/main/java/com/epam/jdi/bdd/JDISteps.java:17]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.MatcherAssert'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/complex/ChecklistAssert.java [hunk 10]
```diff
@@ -8,6 +8,7 @@
 import java.util.List;
 
 import static com.epam.jdi.light.asserts.core.SoftAssert.*;
+import static org.hamcrest.Matchers.*;
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/complex/ChecklistAssert.java:11]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/driver/WebDriverFactory.java [hunk 11]
```diff
@@ -15,6 +15,7 @@
 import static com.epam.jdi.light.driver.get.DriverData.*;
 import static com.epam.jdi.light.driver.get.DriverTypes.*;
 import static com.epam.jdi.light.settings.JDISettings.*;
+import static com.epam.jdi.light.settings.WebSettings.*;
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/driver/WebDriverFactory.java:18]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.settings.WebSettings'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/TableSteps.java [hunk 12]
```diff
@@ -13,9 +13,10 @@
 import static com.epam.jdi.light.elements.complex.table.TableMatcher.hasValue;
 import static com.epam.jdi.light.elements.complex.table.TableMatcher.*;
 import static com.epam.jdi.light.elements.init.entities.collection.EntitiesCollection.*;
+import static com.epam.jdi.tools.LinqUtils.*;
```
Comment (codacy-review) [inline jdi-bdd/src/main/java/com/epam/jdi/bdd/stepdefs/TableSteps.java:16]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.tools.LinqUtils'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 9
- Added 9
           

Complexity increasing per file
==============================
- jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java  1
- jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/BaseTable.java  13
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565&bid=18266460)

File: jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java (hunk 7 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java:7]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.common.Exceptions'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java (hunk 5 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java:10]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/asserts/TextAreaAssert.java (hunk 2 above)
Comment (codacy-review) [inline jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/asserts/TextAreaAssert.java:13]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/complex/ChecklistAssert.java (hunk 10 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/complex/ChecklistAssert.java:11]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/driver/WebDriverFactory.java (hunk 11 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/driver/WebDriverFactory.java:18]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.settings.WebSettings'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java (hunk 3 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java:8]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.settings.WebSettings'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java (hunk 4 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java:9]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.asserts.core.SoftAssert'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UISelectAssert.java (hunk 1 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UISelectAssert.java:18]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.tools.LinqUtils'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/settings/WebSettings.java [hunk 13]
```diff
@@ -105,9 +106,21 @@ public static String printSmartLocators(IBaseElement el) {
...
+    private static void fillAction(JAction1<String> action, String name) {
+        String prop = null;
+        try {
+            prop = getProperty(name);
+        } catch (Exception ignore) {}
+        if (prop == null) {
+            prop = "null";
+        }
+        logger.debug("fillAction(%s=%s)", name, prop);
+        PropertyReader.fillAction(action, name);
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/settings/WebSettings.java:118]: Codacy Issue found: [Unnecessary use of fully qualified name 'PropertyReader.fillAction' due to existing static import 'com.epam.jdi.tools.PropertyReader.fillAction'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

**Comment by mf-code** [comment]
[5381] - 5M9fsj5v

**Comment by mf-code** [comment]
[5381] - cIvZPJuI

**Comment by mf-code** [comment]
[5381] - 4PQ6VjEx

**Comment by mf-code** [comment]
[5381] - Np2IXNNA

**Comment by mf-code** [comment]
[5381] - NqNv8Trz

**Comment by mf-code** [comment]
[5381] - QV44zG9T

**Comment by mf-code** [comment]
[5381] - Allure report on Netlify: https://5ec7d0b29cce67bc651f386d--musing-swartz-6fc060.netlify.app

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 9
- Added 8
           

Complexity increasing per file
==============================
- jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/BaseTable.java  13
- jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565&bid=18266460)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/complex/ChecklistAssert.java (hunk 10 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/complex/ChecklistAssert.java:11]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UISelectAssert.java (hunk 1 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UISelectAssert.java:18]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.tools.LinqUtils'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java (hunk 7 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java:7]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.common.Exceptions'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/asserts/TextAreaAssert.java (hunk 2 above)
Comment (codacy-review) [inline jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/asserts/TextAreaAssert.java:13]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java (hunk 5 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java:10]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java (hunk 3 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java:8]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.settings.WebSettings'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/driver/WebDriverFactory.java (hunk 11 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/driver/WebDriverFactory.java:18]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.settings.WebSettings'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java (hunk 4 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java:9]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.asserts.core.SoftAssert'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

**Comment by mf-code** [comment]
[5385] - i9fiKJhC

**Comment by mf-code** [comment]
[5385] - LUFqdGvG

**Comment by mf-code** [comment]
[5385] - lghqiQvn

**Comment by mf-code** [comment]
[5385] - YiPwzlcY

**Comment by mf-code** [comment]
[5385] - UwhFEYkJ

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 9
- Added 8
           

Complexity increasing per file
==============================
- jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java  1
- jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/BaseTable.java  13
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565&bid=18266460)

File: jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/asserts/TextAreaAssert.java (hunk 2 above)
Comment (codacy-review) [inline jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/asserts/TextAreaAssert.java:13]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/driver/WebDriverFactory.java (hunk 11 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/driver/WebDriverFactory.java:18]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.settings.WebSettings'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java (hunk 5 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java:10]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java (hunk 4 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UIMSelectAssert.java:9]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.asserts.core.SoftAssert'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UISelectAssert.java (hunk 1 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/UISelectAssert.java:18]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.tools.LinqUtils'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java (hunk 7 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java:7]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.common.Exceptions'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java (hunk 3 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/actions/ActionProcessor.java:8]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.settings.WebSettings'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/complex/ChecklistAssert.java (hunk 10 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/complex/ChecklistAssert.java:11]: Codacy Issue found: [Avoid unused imports such as 'org.hamcrest.Matchers'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5570565)

**Comment by mf-code** [comment]
[5391] - x37bRRKJ

**Comment by mf-code** [comment]
[5391] - 7XPeWFRE

**Comment by mf-code** [comment]
[5391] - kuNmvAC3

**Comment by mf-code** [comment]
[5391] - cRtpaIWa

**Comment by mf-code** [comment]
[5391] - vBGWR0gJ

**Comment by mf-code** [comment]
[5391] - 1vUNjpbm

**Comment by mf-code** [comment]
[5391] - Allure report on Netlify: https://5ecb8398f5ea88d7dfd60d3e--musing-swartz-6fc060.netlify.app

**Pull request merged**